### ISSUE TITLE
Trouble parsing API result in Rails controller

### ISSUE THREAD

**Issue by jaymac28** [wikiti/scaleapi-ruby#2]
Thanks so much for your work on this — really sped things up for us, but ran into a problem when trying to parse the results. I can get into all the details here, but the Stackoverflow question I posted today might just be quicker.

Basically, we couldn't easily or cleanly integrate the `build_callback` method with a Rails controller, and even the posted solution which explains what was happening caused new problems with Rails strong parameters. You may have a solution that wasn't considered?

[http://stackoverflow.com/questions/41990408/how-to-parse-api-response-in-rails-controller](http://stackoverflow.com/questions/41990408/how-to-parse-api-response-in-rails-controller)

**Comment by wikiti** [comment]
Hello, @jaymac28!

Thanks for your feedback. It seems like there is somethign wrong with the wrapper. Maybe it's related to [this method](https://github.com/wikiti/scaleapi-ruby/blob/master/lib/scale/callbacks/task.rb#L17). I'll try to give it a look on Monday (this weekend I'll be traveling to another city, and I won't have access to my development tools).

Best regards,
Daniel

**Comment by wikiti** [comment]
Hello again, @jaymac28!

I've just published the new version `0.2.1` which should fix the error. The problem was related to a conflict between how Rails handles parameters (`ActionController::Parameters`) and how this library manages them (`ActiveSupport::HashWithIndifferentAccess`).

The previous version of this gem used wrongly the `ActiveSupport::HashWithIndifferentAccess` constructor in [Scale::to_hash](https://github.com/wikiti/scaleapi-ruby/blob/v0.2.0/lib/scale.rb#L37). Now, the bug is fixed with some checks on the same [method](https://github.com/wikiti/scaleapi-ruby/blob/v0.2.1/lib/scale.rb#L37).

If you can't update the gem (or don't want to), you may want to pass a hash instead to `scale.build_callback` by using `ActionController::Parameters#to_unsafe_hash`:

```rb
callback = $scale.build_callback params.to_unsafe_hash, 'task',
                                 callback_key: request.headers['scale-callback-auth']
```

Could you please confirm if it's working properly?

Best regards,
Daniel

**Comment by jaymac28** [comment]
Thank you! Seems to be working for me now.
